xscen/catutils.py

```diff
@@ -162,20 +177,20 @@ def _compile_pattern(pattern: str) -> parse.Parser:
 
 
 def _name_parser(
-    path: os.PathLike,
-    root: os.PathLike,
+    path: Union[os.PathLike, str],
+    root: Union[os.PathLike, str],
     patterns: list[Union[str, parse.Parser]],
     read_from_file: Optional[Union[list[str], dict]] = None,
     attrs_map: Optional[dict] = None,
     xr_open_kwargs: Optional[dict] = None,
-):
+) -> Union[dict, None]:
```

Comment:
Is `Optional[dict]` a valid return signature?

Reply:
I think so. I'll change it.